### PR TITLE
1 2 display media options

### PR DESCRIPTION
**Pull request by ngilain**
[OO-FEATURE] A new twig function is introduced: render_media to render a media with several options
[OO-DEPRECATED] The twig function display_media is deprecated, use render_media instead
[OO-FEATURE] The Open Orchestra display media strategies relies no more on the media mime type but on the media type
[OO-DEPRECATED] The several displayMedia methods (DisplayMediaManager + strategies) are deprecated, user renderMedia instead

### CODE REVIEW

File: Media/DisplayMedia/DisplayMediaManager.php [hunk 1]
```diff
@@ -59,6 +61,9 @@ public function displayPreview(MediaInterface $media)
      */
     public function displayMedia(MediaInterface $media, $format = '', $style = '')
     {
+        @trigger_error('The '.__METHOD__.' method is deprecated since version 1.2.0 and will be removed in 2.0.0.'
```
Comment (alavieille) [inline Media/DisplayMedia/DisplayMediaManager.php]: can you oo tag deprecated for this method ?

File: Media/DisplayMedia/Strategies/ImageStrategy.php [hunk 2]
```diff
@@ -9,33 +9,66 @@
...
+
     /**
      * @param MediaInterface $media
      *
      * @return bool
      */
     public function support(MediaInterface $media)
     {
-        return strpos($media->getMimeType(), 'image') === 0;
+        return self::MEDIA_TYPE == $media->getMediaType();
```
Comment (alavieille) [inline Media/DisplayMedia/Strategies/ImageStrategy.php]: ===

File: Media/DisplayMedia/Strategies/PdfStrategy.php [hunk 3]
```diff
@@ -18,30 +18,59 @@ class PdfStrategy extends AbstractStrategy
      */
     public function support(MediaInterface $media)
     {
-        return self::MIME_TYPE_PDF == $media->getMimeType();
+        return self::MEDIA_TYPE == $media->getMediaType();
```
Comment (alavieille) [inline Media/DisplayMedia/Strategies/PdfStrategy.php]: `===`

File: Media/DisplayMedia/Strategies/VideoStrategy.php [hunk 4]
```diff
@@ -18,30 +31,83 @@ class VideoStrategy extends AbstractStrategy
      */
     public function support(MediaInterface $media)
     {
-        return strpos($media->getMimeType(), self::MIME_TYPE_FRAGMENT_VIDEO) === 0;
+        return self::MEDIA_TYPE == $media->getMediaType();
```
Comment (alavieille) [inline Media/DisplayMedia/Strategies/VideoStrategy.php]: `===`

File: MediaBundle/Resources/views/RenderMedia/default.html.twig [hunk 5]
```diff
@@ -0,0 +1,15 @@
+<a
+    href="{{ media_url }}"
+    target="_blank"
+    {% if id -%}
```
Comment (alavieille) [inline MediaBundle/Resources/views/RenderMedia/default.html.twig]: why use {% if id -%} and not {% if id %} ?

Comment (ngilain) [inline MediaBundle/Resources/views/RenderMedia/default.html.twig]: the - remove the empty spaces between the tag and the html following

**Comment by ngilain** [comment]
@alavieille : update done

**Comment by alavieille** [comment]
:+1:

File: Media/DisplayMedia/Strategies/AbstractStrategy.php [hunk 6]
```diff
@@ -52,6 +57,102 @@ public function displayPreview(MediaInterface $media)
...
+            . ' Implement it in the strategies instead.', E_USER_DEPRECATED);
+
+        $format = '';
+        if (isset($options['format'])) {
+            $format = $options['format'];
+        }
+
+        $style = '';
+        if (isset($options['style'])) {
+            $format = $options['style'];
```
Comment (itkg-canne) [inline Media/DisplayMedia/Strategies/AbstractStrategy.php]: There is a mistake here. You should define value of `$style` var.

Comment (alavieille) [inline Media/DisplayMedia/Strategies/AbstractStrategy.php]: @itkg-canne https://github.com/open-orchestra/open-orchestra-media-bundle/pull/226

**Pull request merged**